allennlp/data/vocabulary.py

```diff
@@ -406,6 +407,112 @@ def from_params(cls, params: Params, instances: Iterable['adi.Instance'] = None)
                                          only_include_pretrained_words=only_include_pretrained_words,
                                          tokens_to_add=tokens_to_add)
 
+    def extend(self,
+               counter: Dict[str, Dict[str, int]] = None,
+               min_count: Dict[str, int] = None,
+               max_vocab_size: Union[int, Dict[str, int]] = None,
+               non_padded_namespaces: Sequence[str] = DEFAULT_NON_PADDED_NAMESPACES,
+               pretrained_files: Optional[Dict[str, str]] = None,
+               only_include_pretrained_words: bool = False,
+               tokens_to_add: Dict[str, List[str]] = None) -> None:
+        """
+        This method can be used for extending already generated vocabulary.
+        It takes same parameters as Vocabulary initializer. The token2index
+        and indextotoken mappings of calling vocabulary will be retained.
+        It is an inplace operation so None will be returned.
+        """
+        if not isinstance(max_vocab_size, dict):
+            int_max_vocab_size = max_vocab_size
+            max_vocab_size = defaultdict(lambda: int_max_vocab_size)  # type: ignore
+        min_count = min_count or {}
+        pretrained_files = pretrained_files or {}
+
+        if counter is not None:
+            # Make sure vocabulary extension is safe.
+            # Common named namespaces must have same setting for padded=True/False
+            tob_added_namespaces = list(counter.keys())
+            existing_namespaces = self.get_all_namespaces()
+            for namespace in tob_added_namespaces:
+                if namespace in existing_namespaces:
+                    existing_namespace_padded = namespace in self._non_padded_namespaces
+                    tob_added_namespace_padded = namespace in non_padded_namespaces
```

Comment:
I don't like these variable names, I would call them `original_padded` and `extension_padded` or something like that

Reply:
Sure

Reply:
This block is cleaned up now.